tests/common/pkt_filter/filter_pkt_in_buffer.py

```diff
@@ -0,0 +1,232 @@
+from collections import OrderedDict
+
+import time
+import json
+import copy
+
+import ptf.mask as mask
+import ptf.packet as packet
+
+
+NATIVE_TYPE = (int, float, long, bool, list, dict, tuple, set, str, bytes, unicode, type(None))
+
+
+def _parse_layer(layer):
+    """
+    Convert packet layer to dictionary
+
+    Args:
+        layer: Layer of packet
+
+    Returns:
+        Field dictionary
+    """
+    fields = {}
+
+    if not hasattr(layer, 'fields_desc'):
+        return None
+
+    for field in layer.fields_desc:
+        value = getattr(layer, field.name)
+        if value is type(None):
+            value = None
+
+        if not isinstance(value, NATIVE_TYPE):
+            value = _parse_layer(value)
+
+        fields[field.name] = str(value)
+
+    return {layer.name: fields}
+
+
+def convert_pkt_to_dict(pkt):
+    """
+    Convert scapy packet to dictionary
+
+    Args:
+        pkt: Scapy packet
+
+    Returns:
+        Packet dictionary
+    """
+    packet_dict = OrderedDict()
+    counter = 0
+
+    while True:
+        layer = pkt.getlayer(counter)
+        if not layer:
+            break
+
+        layer_dict = _parse_layer(layer)
+        if layer_dict:
+            packet_dict.update(layer_dict)
+
+        counter += 1
```

Comment:
What's the purpose of this `counter` variable?

Reply:
This variable is used for [getting ](https://github.com/Azure/sonic-mgmt/blob/fb054c313402b8b03530f319b883ba7c953a46e0/tests/common/pkt_filter/filter_pkt_in_buffer.py#L56)fields of packet by using field number.